### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by Roksolanka**
Added profile model and view. Added view FileManager.

### CODE REVIEW

File: events/events/urls.py [hunk 1]
```diff
@@ -24,4 +24,5 @@
     url(r'^auth/', include('auth.urls', namespace='auth')),
     url(r'^api/v1/events/', include('event.urls', namespace='event')),
     url(r'^api/v1/reg/', include('registration.urls', namespace='reg')),
+    url(r'^profile/', include('profiles.urls', namespace='profiles')),
```
Comment (Scandie) [inline events/events/urls.py]: @Roksolanka please use `api/v1/profile/`

**Review by Scandie — CHANGES_REQUESTED** [review]

File: events/profiles/admin.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from django.contrib import admin
+from models import UserProfile
+
+
+admin.site.register(UserProfile)
```
Comment (Scandie) [inline events/profiles/admin.py]: @Roksolanka add empty line at the end of the file.

File: events/profiles/models.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    photo = models.CharField(max_length=200, default='default_photo.png')
+    education = models.TextField(blank = True, default='')
+    job = models.TextField(blank = True, default='')
+
+    @classmethod
+    def create_user_profile(cls, **kwargs):
+        UserProfile.objects.get_or_create(user=kwargs['user'])
+
+    @classmethod
+    def get_all(cls):
```
Comment (yevgenbehen) [inline events/profiles/models.py]: No need in this method.

File: events/events/urls.py (hunk 1 above)
Comment (yevgenbehen) [reply events/events/urls.py]: +1 to @Scandie comment

File: events/profiles/views.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                    profile.save()
+                    return JsonResponse({'status': 'success'}, status=200)
+		else:
+		    return JsonREsponse({'status': 'Some fields are incorrect. Please check.'}, status=400)
+            except:
+        	   return JsonResponse({'status': 'Please, check your personal information'}, status=400)
+        else:
+            return HttpResponseForbidden('Permission denied')
+
+    def __get_dict(self, profile):
```
Comment (yevgenbehen) [inline events/profiles/views.py]: This method does not get dict - it gets JSON. Please rename method.

File: events/profiles/views.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+import json
+import boto3
```
Comment (yevgenbehen) [inline events/profiles/views.py]: Why do you need boto3? I do not see it's usage in code

File: requirements.txt [hunk 6]
```diff
@@ -2,3 +2,7 @@ Django==1.10.2
 django-static-precompiler==1.5
 psycopg2==2.6.2
 validate-email==1.3
+boto==2.43.0
+boto3==1.4.1
+botocore==1.4.65
```
Comment (yevgenbehen) [inline requirements.txt:7]: Where do you use this?

File: requirements.txt [hunk 7]
```diff
@@ -2,3 +2,7 @@ Django==1.10.2
...
+s3transfer==0.1.9
```
Comment (yevgenbehen) [inline requirements.txt]: The same question here.

File: events/profiles/views.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                key_bucket.set_contents_from_filename(path['path'])
+                return JsonResponse({'key_photo': key_bucket.key})
+            except:
+                return JsonResponse({'status': 'Please check your path : ' + path['path']}, status=400)
+        else:
+            return JsonResponse({'status': 'Please, input correct path'}, status=400)
+
+    @classmethod
+    def get_href(key, bucket_name=local_settings.NAME_PROFILES_BUCKET):
+        conn = S3Connection(local_settings.ACCESS_KEY_ID, settings.SECRET_KEY)
```
Comment (yevgenbehen) [inline events/profiles/views.py]: You can use private method `__connect_to_S3` here

File: events/profiles/views.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        return conn.generate_url(300, "GET", bucket_name, key)
```
Comment (yevgenbehen) [inline events/profiles/views.py]: @Roksolanka you have told on today's meeting that it does not work for you and you can't connect to S3 second time. Could you please be more specific? What error did you get and please provide more details about circumstances? Thank you!

File: events/profiles/views.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        if request.user.is_authenticated:
+            info = json.loads(request.body.decode()) 
+            try:
+                profile = UserProfile()
+                profile.user = User.get_user_by_id(info.get('user'))
+                profile.photo = info.get('photo')
+                profile.education = info.get('education')
+                profile.job = info.get('job')
+                form = ProfileForm(profile)
+                if form.is_valid():
```
Comment (BorysYudin) [inline events/profiles/views.py]: It would be better validate data at first and then assign it to profile fields.

**Review by Scandie — APPROVED** [review]

**Pull request merged**